graylog2-server/src/main/java/org/graylog2/streams/PaginatedStreamService.java

```diff
@@ -57,20 +58,28 @@ public long count() {
     public PaginatedList<StreamDTO> findPaginated(SearchQuery searchQuery, Predicate<StreamDTO> filter, int page,
                                                   int perPage, String sortField, String order) {
         final Bson dbQuery = searchQuery.toBson();
-        final DBSort.SortBuilder sortBuilder = getSortBuilder(order, sortField);
+        var pipelineBuilder = ImmutableList.<Bson>builder()
+                .add(Aggregates.match(dbQuery));
 
-        final AggregateIterable<Document> result = collection.aggregate(List.of(
-                Aggregates.match(dbQuery),
-                Aggregates.lookup(
-                        MongoIndexSetService.COLLECTION_NAME,
-                        List.of(new Variable<>("index_set_id", new Document("$toObjectId", "$index_set_id"))),
-                        List.of(Aggregates.match(doc("$expr", doc("$eq", List.of("$_id", "$$index_set_id"))))),
-                        "index_set"
-                ),
-                Aggregates.set(new Field<>("index_set_title", new Document("$first", "$index_set.title"))),
-                Aggregates.project(doc("index_set", 0)),
-                Aggregates.sort(sortBuilder)
-        ));
+        if (sortField.equals("index_set_title")) {
+            pipelineBuilder.add(Aggregates.lookup(
+                            MongoIndexSetService.COLLECTION_NAME,
+                            List.of(new Variable<>("index_set_id", doc("$toObjectId", "$index_set_id"))),
+                            List.of(Aggregates.match(doc("$expr", doc("$eq", List.of("$_id", "$$index_set_id"))))),
+                            "index_set"
+                    ))
+                    .add(Aggregates.set(new Field<>("index_set_title", doc("$first", "$index_set.title"))));
+        }
+
+        if (isStringField(sortField)) {
+            pipelineBuilder.add(Aggregates.set(new Field<>("lower" + sortField, doc("$toLower", "$" + sortField))))
+                    .add(Aggregates.sort(getSortBuilder(order, "lower" + sortField)))
+                    .add(Aggregates.project(Projections.exclude("index_set", "lower" + sortField)));
+        } else {
+            pipelineBuilder.add(Aggregates.sort(getSortBuilder(order, sortField)));
```

Comment:
maybe a `Projections.exclude("index_set"` missing here?

Reply:
I decided to drop it, because we do not really need to strip it out (it will just not be used when creating the `StreamDTO`) and it adds yet another branch (because we would need another `if (sortField.equals("index_set_title") {...}`. I could remove it from the previous `Projections.exclude` too though, for consistency.